notify/impl.go

```diff
@@ -934,6 +938,79 @@ func (n *Pushover) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
 	return false, nil
 }
 
+type Jira struct {
+	conf *config.JiraConfig
+	tmpl *template.Template
+}
+
+func NewJira(c *config.JiraConfig, t *template.Template) *Jira {
+	log.Debugf("template: %#v", *t)
+	return &Jira{conf: c, tmpl: t}
+}
+
+type jiraCreateIssueRequest struct {
+	Fields map[string]interface{} `json:"fields"`
+}
+
+type jiraCreateIssueResponse struct {
+	ID   string `json:"id"`
+	Key  string `json:"key"`
+	Self string `json:"self"`
+}
+
+// Notify implements the Notifier interface.
+func (n *Jira) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	var err error
+	data := n.tmpl.Data(receiverName(ctx), groupLabels(ctx), as...)
+	tmpl := tmplText(n.tmpl, data, &err)
+
+	fields := map[string]interface{}{
+		"project":     map[string]string{"key": tmpl(n.conf.Project)},
+		"issuetype":   map[string]string{"name": tmpl(n.conf.IssueType)},
+		"summary":     tmpl(n.conf.Summary),
+		"description": tmpl(n.conf.Description),
+	}
+	// check errors from tmpl()
```

Comment:
`tmplText()` returns a function which I'm storing in `tmpl`. the `tmpl` function, when called, updates the err object passed into `tmplText`. So i'm checking the result of the `tmpl` calls here, not `tmplText`.